src/components/webview/FileExplorerProvider.ts

```diff
@@ -540,6 +558,16 @@ export class FileExplorerProvider implements WebviewViewProvider {
 			commands.executeCommand('intuita.disposeView', message.webviewName);
 		}
 
+		if (message.kind === 'webview.fileExplorer.setState') {
+			this.__workspaceState.setFocusedFileExplorerNodeId(
+				message.focusedId,
+			);
+
+			this.__workspaceState.setOpenedFileExplorerNodeIds(
+				new Set(message.openedIds),
+			);
```

Comment:
should this trigger rerender?

Reply:
in general it should, but let's do the rerender once we've moved to the new state management